corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -131,7 +144,11 @@ def serial_processors(self):
         else:
             return self.processors
 
-    def process_changes(self, since, forever):
+    def wait_for_change(self, since):
+        """Hang until there is another change to process from the feed"""
+        next(self.get_change_feed().iter_changes(since=since or None, forever=True))
+
+    def process_changes(self, since, forever=False):
```

Comment:
I don't think this is ever called with `forever = True` now

Reply:
Yeah I was also thinking that. I wanted to first see your reaction to the change before crossing the moat and pulling up the drawbridge, but it seems like you've had similar issues on ICDS and are willing to adopt this new strategy there as well (provided of course that it's well-tested)?

Reply:
yes, based on your description I think this is a general issue